src/py/flwr/server/superlink/state/sqlite_state.py

```diff
@@ -704,29 +779,34 @@ def get_node_public_keys(self) -> Set[bytes]:
 
     def get_run(self, run_id: int) -> Optional[Run]:
         """Retrieve information about the run with the specified `run_id`."""
+        sint64_run_id = convert_uint64_to_sint64(run_id)
         query = "SELECT * FROM run WHERE run_id = ?;"
         try:
-            row = self.query(query, (run_id,))[0]
+            row = self.query(query, (sint64_run_id,))[0]
             return Run(
-                run_id=run_id,
+                run_id=convert_sint64_to_uint64(row["run_id"]),
                 fab_id=row["fab_id"],
                 fab_version=row["fab_version"],
                 fab_hash=row["fab_hash"],
                 override_config=json.loads(row["override_config"]),
             )
         except sqlite3.IntegrityError:
-            log(ERROR, "`run_id` does not exist.")
+            log(ERROR, "`sint64_run_id` does not exist.")
```

Comment:
I think it's not neccessary to mention it's sint64 here as it's just an implementation detail. This might confuse users who host their own SuperLink.

```suggestion
            log(ERROR, "`run_id` does not exist.")
```